### PR TITLE
Use several NAV files when observations are over several days

### PR DESCRIPTION
**Pull request by plutonheaven**
- Changed `nav_file_discovery` to provide several NAV files for observations spanning over several days.
- Changed `main._build_records_cached` to iterate over several NAV files for sat states and iono delay computation
- Added test with a first observation at midnight

### CODE REVIEW

**Comment by plutonheaven** [comment]
@jtec, can you test this PR on Mac?

The current version fails on Windows, because `gfzrnx` is not executed correctly. I'll investigate this in a separate branch.

**Comment by jtec** [comment]
@plutonheaven Ran `pytest` on 56a5ebd4cf90fda37564052c87bd3a51eb1d0dbf, everything passes.

**Comment by plutonheaven** [comment]
This PR seems to work.
Take some time to review the code though, having several NAV files complexifies a bit the code because you have to choose the right file depending on the date. E.g. for the iono delay, the Klobuchar parameters are collected in a dictionnary, because they are valid only for a single day...

**Comment by jtec** [comment]
@plutonheaven Do you know whether there is a tool that can merge rinex nav files? We could then use that and use the resulting file as-is.

**Comment by plutonheaven** [comment]
`gfzrnx` can do it. But I don't know what happens to daily parameters that are stored in a RINEX NAV file header, such as the SYSTEM TIME CORR or IONO CORR.

I will do some tests to see what happens. Let's wait for those results before approving this PR.

**Comment by plutonheaven** [comment]
Merging several RINEX NAV is performed by `gfzrnx`. However, when doing this in RINEX-3, only one set of header parameters is kept, and the logic behind which one is kept is not clear.

This changes in RINEX-4 format, because ION corrections are dedicated 'payload' messages, no longer in the header. But right now, `georinex` does not parse RNX4 files.

I think the solution proposed in this PR does not discard any header information for multiple RINEX NAV files, and may be a good solution for now.

File: src/prx/rinex_nav/nav_file_discovery.py [hunk 1]
```diff
@@ -16,15 +19,28 @@ def is_rinex_3_mixed_mgex_broadcast_ephemerides_file(file: Path):
 
 
 def try_downloading_ephemerides_from_bkg(
+    # This function returns a list of paths, in case the observation file spans over several days
```
Comment (jtec) [inline src/prx/rinex_nav/nav_file_discovery.py:22]: nit: "spans several days"?

File: src/prx/rinex_nav/nav_file_discovery.py [hunk 2]
```diff
@@ -16,15 +19,28 @@ def is_rinex_3_mixed_mgex_broadcast_ephemerides_file(file: Path):
...
+        [
+            pd.Timestamp(year=t_start.year,month=t_start.month,day=t_start.day),
+            pd.Timestamp(year=t_end.year,month=t_end.month,day=t_end.day)
+        ],
+    )
+    for time in time_starts:
         # IGS BKG Rinex 3.04 mixed file paths follow this pattern:
+        # For files after 2022
         # https://igs.bkg.bund.de/root_ftp/IGS/BRDC/2023/002/BRDC00IGS_R_20230020000_01D_MN.rnx.gz
+        # For files before 2022
```
Comment (jtec) [inline src/prx/rinex_nav/nav_file_discovery.py:36]: nice!

**Review by jtec — APPROVED** [review]

**Pull request merged**